spec/http.md

```diff
@@ -0,0 +1,51 @@
+HTTP
+====
+
+## Resources
+
+Each EC2 instance MUST present the following (GET) resources:
+
+    [/service/healthcheck](#healthcheck)
+    [/service/gtg](#gtg)
+    [/service/dependencies](#dependencies)
+
+These resources should be presented on port 8080 (not port 80).
+
+### <a name="healthcheck">Healthcheck</a>
+
+Is the service running? Two possible responses:
+
+    200 "OK" (service is running)
+    50* (service is down - choose exact status code as appropriate)
+
+### <a name="gtg">GTG</a>
+
+    200 "OK" (service is able to usefully respond to requests)
+    50* (service is down - choose exact status code as appropriate)
+
+Is the service able to usefully respond to requests? (Typically, this
+indicates whether the service is running and its dependencies are
+available as expected.)
```

Comment:
Note that https://github.com/beamly/SE4/blob/1.0/SE4.md defines the concepts of "Healthy" and "GTG" differently. In their definition:

* Healthy = I'm OK and all of my dependencies are OK
* GTG = I can receive traffic (but I might not be healthy)

I'm not sure of the value of having 2 endpoints like this. I think it would just lead to confusion about which one you're supposed to call. 

I would have one endpoint called `healthcheck` that returns a 200 "OK" if all of the following are true:

* the service itself is healthy
* all of its dependencies are healthy
* the service can respond to requests (this is implied by the fact that it responded to the healthcheck)